lib/faraday/connection.rb

```diff
@@ -163,8 +163,16 @@ def headers=(hash)
     METHODS_WITH_QUERY.each do |method|
       class_eval <<-RUBY, __FILE__, __LINE__ + 1
         def #{method}(url = nil, params = nil, headers = nil)
+          if params
+            headers ||= {}
+            if headers[:url_params]
+              headers[:url_params] = params.update(headers[:url_params])
+            else
+              headers[:url_params] = params
+            end
+          end
```

Comment:
So what happens if I want to specify a `body` without using the `params` argument?
```ruby
conn.get('?p=1&up=1&a=a', body: 'body') 
```

My impression is that the above won't work because we always expect `body` and `url_params` to be provided only through `headers`

Reply:
Ah, just pass `nil`: `conn.get('?p=1&up=1&a=a', nil, body: 'body')`